sources/web/datalab/polymer/components/file-browser/file-browser.ts

```diff
@@ -452,9 +463,11 @@ class FileBrowserElement extends Polymer.Element implements DatalabPageElement {
           Utils.log.error(msgPrefix, e);
         }
       })
+      // TODO: This is not very accurate if multiple fetch requests are running
```

Comment:
There are still a bunch of places in the code where it does not take action if this._fetching is set. I think it will be pretty easy to get into the state where this flag is cleared while something is still running, which means your checks won't work properly and things will execute while a fetch is running. If this is a not a problem, then you should remove those checks; if it is a problem, then you need to make _fetching more accurate.